rasa/shared/nlu/training_data/training_data.py

```diff
@@ -861,6 +880,51 @@ def _to_tf_features(self, features: List[Features]) -> Dict[Text, tf.train.Featu
 
         return tf_features
 
+    @staticmethod
+    def _relevant_message_data_keys() -> List[Text]:
+        from rasa.nlu.constants import TOKENS_NAMES
+
+        return [
+            TEXT,
+            INTENT,
+            RESPONSE,
+            INTENT_RESPONSE_KEY,
+            ENTITIES,
+            TOKENS_NAMES[TEXT],
+        ]
+
+    def _encode_message_data(
```

Comment:
Is this the right place for this? Where do we normally encode training data?

Reply:
We did not encode the message data before. (Maybe encode is not the best word.) We need to write down the training data incl. features to files. As we also need to have the text and labels of the message, we need to "encode" the data into tf.Features.